samples/python/agents/langgraph/agent.py

```diff
@@ -60,9 +73,48 @@ class CurrencyAgent:
         "Set response status to error if there is an error while processing the request."
         "Set response status to completed if the request is complete."
     )
-     
+
     def __init__(self):
-        self.model = ChatGoogleGenerativeAI(model="gemini-2.0-flash")
+        if os.getenv("GOOGLE_API_KEY"):
```

Comment:
Let's use switch case instead of multiple if-else statements ?

Reply:
Done. 
Python>=3.13 is a prerequisite so we can assume switch statements are available.